### PR TITLE
[lib & admin & template  & plugins] Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8')

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #10399 .
#### Summary of Changes

Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8')
#### Testing Instructions

Please do a code review or test all changed files.

### CODE REVIEW

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 6aa2082258b0a1ff586f4343fb6387ede2cfcd3e<br /><br />on review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 6aa2082258b0a1ff586f4343fb6387ede2cfcd3e<br /><br />on review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

File: libraries/joomla/document/opensearch.php [hunk 1]
```diff
@@ -133,11 +133,11 @@ public function render($cache = false, $params = array())
 		$elOs = $xml->createElementNs($osns, 'OpenSearchDescription');
 
 		$elShortName = $xml->createElementNs($osns, 'ShortName');
-		$elShortName->appendChild($xml->createTextNode(htmlspecialchars($this->_shortName)));
+		$elShortName->appendChild($xml->createTextNode(htmlspecialchars($this->_shortName), ENT_COMPAT, 'UTF-8'));
```
Comment (andrepereiradasilva) [inline libraries/joomla/document/opensearch.php]: this is not correct!

File: libraries/joomla/document/opensearch.php [hunk 2]
```diff
@@ -133,11 +133,11 @@ public function render($cache = false, $params = array())
...
 		$elOs->appendChild($elShortName);
 
 		$elDescription = $xml->createElementNs($osns, 'Description');
-		$elDescription->appendChild($xml->createTextNode(htmlspecialchars($this->description)));
+		$elDescription->appendChild($xml->createTextNode(htmlspecialchars($this->description), ENT_COMPAT, 'UTF-8'));
```
Comment (andrepereiradasilva) [inline libraries/joomla/document/opensearch.php]: this is not correct!

File: libraries/joomla/document/opensearch.php [hunk 3]
```diff
@@ -151,7 +151,7 @@ public function render($cache = false, $params = array())
 			$elImage->setAttribute('type', $image->type);
 			$elImage->setAttribute('width', $image->width);
 			$elImage->setAttribute('height', $image->height);
-			$elImage->appendChild($xml->createTextNode(htmlspecialchars($image->data)));
+			$elImage->appendChild($xml->createTextNode(htmlspecialchars($image->data), ENT_COMPAT, 'UTF-8'));
```
Comment (andrepereiradasilva) [inline libraries/joomla/document/opensearch.php]: this is not correct!

File: libraries/joomla/document/opensearch.php (hunk 1 above)
Comment (brianteeman) [inline libraries/joomla/document/opensearch.php]: Why not?

**Comment by andrepereiradasilva** [comment]
@brianteeman please remove the RTC from this one. It as errors.

**Comment by brianteeman** [comment]
Please explain the errors - saying it is not correct is not helpful

**Comment by andrepereiradasilva** [comment]
`htmlspecialchars($image->data), ENT_COMPAT, 'UTF-8'`

**Comment by brianteeman** [comment]
Explain not quote ;)

**Comment by andrepereiradasilva** [comment]
the parentisis is the wrong place

**Comment by andrepereiradasilva** [comment]
travis is also telling that.

**Comment by brianteeman** [comment]
RTC removed as requested<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

File: libraries/joomla/document/opensearch.php (hunk 1 above)
Comment (andrepereiradasilva) [inline libraries/joomla/document/opensearch.php]: ´$xml->createTextNode(htmlspecialchars($this->_shortName), ENT_COMPAT, 'UTF-8')`

should be

`$xml->createTextNode(htmlspecialchars($this->_shortName, ENT_COMPAT, 'UTF-8'))`

**Comment by wojsmol** [comment]
@zero-24 see https://github.com/zero-24/joomla-cms/issues/12

**Comment by andrepereiradasilva** [comment]
seems all fine to me now.

**Comment by zero-24** [comment]
Thanks @wojsmol & @andrepereiradasilva

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 22ff770e35d41f7b1c6309b1efc013c43d9d426b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 22ff770e35d41f7b1c6309b1efc013c43d9d426b<br /><br />on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10438">issues.joomla.org/joomla-cms/10438</a>.</sub>

**Comment by rdeutz** [comment]
@zero-24 could you have a look at the merge conflicts, thanks.

**Comment by zero-24** [comment]
i did broken my branch please remove the milestone. @brianteeman

**Pull request closed without merging**